scrapy/pipelines/images.py

```diff
@@ -126,16 +126,48 @@ def get_images(self, response, request, info):
         if width < self.min_width or height < self.min_height:
             raise ImageException("Image too small (%dx%d < %dx%d)" %
                                  (width, height, self.min_width, self.min_height))
+        
+        def _is_convert_image_overriden():
+            import inspect
+            if six.PY2:
+                convert_image_signature = inspect.getargspec(self.convert_image)
+            elif six.PY3:
+                convert_image_signature = inspect.getfullargspec(self.convert_image)
+            if 'response_body' not in convert_image_signature.args:
+                return True
+            return False
 
-        image, buf = self.convert_image(orig_image)
+        def _warn():
+            from scrapy.exceptions import ScrapyDeprecationWarning
+            import warnings
+            warnings.warn('ImagesPipeline.convert_image() method overriden in a incompatible way, '
+                          'overriden method does not accept response_body argument.',
```

Comment:
You are actually making quite an effort to make the old way work, so I would say “deprecated” instead of “incompatible“ here :slightly_smiling_face: 

Also applied to the other warning below.